app/app_iot_hub.go

```diff
@@ -284,3 +288,51 @@ func (a *app) SetDeviceStateIoTHub(
 	}
 	return a.GetDeviceStateIoTHub(ctx, deviceID, integration)
 }
+
+func (app *app) VerifyDeviceTwin(ctx context.Context, req model.PreauthRequest) error {
+	integrations, err := app.GetIntegrations(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve integration: %w", err)
+	}
+	var integration model.Integration
+	for _, integration = range integrations {
+		if integration.Provider == model.ProviderIoTHub {
+			break
+		}
+	}
+	deviceModule := strings.SplitN(req.DeviceID, "/", 2)
+	for i := range deviceModule {
+		deviceModule[i] = url.PathEscape(deviceModule[i])
+	}
+	id := strings.Join(deviceModule, "/modules/")
+	log.FromContext(ctx).Warnf("getting twin: %s", id)
```

Comment:
```suggestion
```
This looks like a leftover debug log sent to `warn`.

Reply:
fixed